libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts

```diff
@@ -346,6 +362,24 @@ export class UploadCollectionComponent {
             });
     }
 
+    /** @hidden */
+    _checkName(e: FocusEvent, currentItem: UploadCollectionItem): void {
+        const input = e.target as HTMLInputElement;
+        const itemName = input.value.trim();
+        const newName = currentItem.type === 'file' ? `${itemName}.${currentItem.name.split('.').pop()}` : itemName;
+
+        if (
+            !this.sameFileNameAllowed &&
+            this._getList().some((item) => currentItem.documentId !== item.documentId && item.name === newName)
```

Comment:
I think that it is better to write the result of complex processing into a local variable and use it inside `if`.